File: binance4j-wallet/src/main/java/com/binance4j/wallet/dto/ConvertibleAssets.java
package com.binance4j.wallet.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Assets that can be converted into BNB.
 * 
 * @param details            Details.
 * @param totalTransferBtc   BTC valuation.
 * @param totalTransferBNB   BNB valuation.
 * @param dribbletPercentage Commission fee.
 */
@ApiModel("Assets that can be converted into BNB.")
public record ConvertibleAssets(@ApiModelProperty("Details.") List<ConvertibleAsset> details, @ApiModelProperty("BTC valuation.") String totalTransferBtc,
		@ApiModelProperty("BNB valuation.") String totalTransferBNB, @ApiModelProperty("Commission fee.") String dribbletPercentage) {
}

File: connectors/src/main/kotlin/com/binance4j/wallet/dto/ConvertibleAssets.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.wallet.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Assets that can be converted into BNB.
 *
 * @property details Details.
 * @property totalTransferBtc BTC valuation.
 * @property totalTransferBNB BNB valuation.
 * @property dribbletPercentage Commission fee.
 */
@ApiModel("Assets that can be converted into BNB.")
data class ConvertibleAssets(
	@ApiModelProperty("Details.") var details: List<ConvertibleAsset> = emptyList(),
	@ApiModelProperty("BTC valuation.") val totalTransferBtc: String,
	@ApiModelProperty("BNB valuation.") val totalTransferBNB: String,
	@ApiModelProperty("Commission fee.") val dribbletPercentage: String
)
